analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -433,6 +433,48 @@ public void Branching_LearnsObjectConstraint_TypePattern_Negated(string expressi
                 .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
         }
 
+        [DataTestMethod]
+        [DataRow("arg is object o")]
+        [DataRow("arg is string s")]
+        [DataRow("arg is not not object o")]
+        [DataRow("arg is not not string s")]
+        [DataRow("!(arg is not object o)")]
+        public void Branching_LearnsObjectConstraint_DeclarationPattern(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.DeclarationPattern);
+            validator.ValidateTag("If", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
+            validator.ValidateTag("Else", x => x.Should().BeNull("it could be null or any other type"));
+            validator.TagValues("End").Should().HaveCount(2)
+                .And.ContainSingle(x => x == null)
+                .And.ContainSingle(x => x != null && x.HasConstraint(ObjectConstraint.NotNull));
+        }
+
+        [DataTestMethod]
+        [DataRow("arg is var o")]
+        [DataRow("arg is int i")]
+        public void Branching_LearnsObjectConstraint_DeclarationPattern_NoConstraints(string expression)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.DeclarationPattern);
+            validator.ValidateTag("If", x => x.Should().BeNull());
+            validator.ValidateTag("Else", x => x.Should().BeNull());
+            validator.ValidateTag("End", x => x.Should().BeNull());
+        }
+
+        [DataTestMethod]
+        [DataRow("arg is not object o")]
+        [DataRow("arg is not string s")]
+        [DataRow("!(arg is object o)")]
+        [DataRow("!(arg is string s)")]
```

Comment:
Are we planning to support some simple operators on the value itself too, like `someString + "." is "Hello World."` or `!` for nullable bools and unary `-` for `int?`?